### PR TITLE
Prepare for clang 11

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Prepare for build with clang 11.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
How to install clang-11:

```
git clone git@github.com:llvm/llvm-project.git
cd llvm-project
git checkout release/11.x
cd ..
mkdir build
cd build
cmake -GNinja -DCMAKE_BUILD_TYPE:STRING=Release -DLLVM_ENABLE_PROJECTS=all ../llvm-project/llvm/
ninja
sudo ninja install
hash clang
```

**Comment by alexey-milovidov** [comment]
Integration tests are broken in https://github.com/ClickHouse/ClickHouse/pull/14323

File: src/Functions/GatherUtils/Sources.h [hunk 1]
```diff
@@ -122,9 +122,15 @@ struct NumericArraySource : public ArraySourceImpl<NumericArraySource<T>>
     }
 };
 
+
+/// The method can be virtual or not depending on the template parameter.
```
Comment (abyss7) [inline src/Functions/GatherUtils/Sources.h:126]: What method? And what is an original issue with overrides?

Comment (alexey-milovidov) [reply src/Functions/GatherUtils/Sources.h:126]: The methods of ISource.

It isn't possible to specify `override` if the method is not virtual in the base class.

Comment (abyss7) [reply src/Functions/GatherUtils/Sources.h:126]: Now I see.

**Comment by alexey-milovidov** [comment]
AST fuzzer: https://github.com/ClickHouse/ClickHouse/issues/13281

**Pull request merged**